prober/tcp_test.go

```diff
@@ -246,6 +246,46 @@ func TestProbeTCPStartTLSSMTP(t *testing.T) {
 	checkTLSVersionMetrics("TLS 1.3", registry, t)
 }
 
+// TestProbeTCPStartTLSSMTP tests STARTTLS against a mock SMTP server
+// which provides STARTTLS as last option in the list
+func TestProbeTCPStartTLSSMTPLastOptionStartTLS(t *testing.T) {
```

Comment:
I think this name is a little deceiving. If I understand it correctly, it's not really the order in the list of options that is pertinent here. It's the use of a dash vs a space in the STARTTLS command?

Perhaps a better name would be `TestProbeTCPStartTLSSMTPDashInCommand`?